src/finufft_core.cpp

```diff
@@ -590,8 +590,8 @@ FINUFFT_PLAN_T<TF>::FINUFFT_PLAN_T(int type_, int dim_, const BIGINT *n_modes, i
 
 #ifdef _OPENMP
   // choose overall # threads...
-  int ompmaxnthr = MY_OMP_GET_MAX_THREADS();
-  int nthr       = ompmaxnthr; // default: use as many as OMP gives us
+  int ompmaxnthr = getPhysicalCoreCount();
+  int nthr       = ompmaxnthr; // default: use as many physical cores as possible
   // (the above could be set, or suggested set, to 1 for small enough problems...)
   if (opts.nthreads > 0) {
     nthr = opts.nthreads; // user override, now without limit
```

Comment:
(Sorry, I couldn't figure out how to annotate the `fprintf` statement directly...)

The warning statement a few lines below will need adjustment, since we are no longer relying on the number of threads that OpenMP reports as available.